api/lib/domain/usecases/reconcile-oidc-user-for-admin.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @typedef {import ('./index.js').OidcAuthenticationService} OidcAuthenticationService
+ * @typedef {import ('./index.js').AuthenticationSessionService} AuthenticationSessionService
+ * @typedef {import ('./index.js').AuthenticationMethodRepository} AuthenticationMethodRepository
+ * @typedef {import ('./index.js').UserRepository} UserRepository
+ * @typedef {import ('./index.js').UserLoginRepository} UserLoginRepository
+ */
+
+import { AuthenticationKeyExpired, DifferentExternalIdentifierError } from '../errors.js';
+import { AuthenticationMethod } from '../models/index.js';
+
+/**
+ * @param {Object} params
+ * @param {string} params.authenticationKey
+ * @param {string} params.email
+ * @param {string} params.identityProvider
+ * @param {OidcAuthenticationService} params.oidcAuthenticationService
+ * @param {AuthenticationSessionService} params.authenticationSessionService
+ * @param {AuthenticationMethodRepository} params.authenticationMethodRepository
+ * @param {UserRepository} params.userRepository
+ * @param {UserLoginRepository} params.userLoginRepository
+ */
+const reconcileOidcUserForAdmin = async function ({
+  authenticationKey,
+  email,
+  identityProvider,
+  oidcAuthenticationService,
+  authenticationSessionService,
+  authenticationMethodRepository,
+  userRepository,
+  userLoginRepository,
+}) {
+  const sessionContentAndUserInfo = await authenticationSessionService.getByKey(authenticationKey);
+  if (!sessionContentAndUserInfo) {
+    throw new AuthenticationKeyExpired();
+  }
+
+  const foundUser = await userRepository.getByEmail(email);
+
+  const authenticationMethods = await authenticationMethodRepository.findByUserId({ userId: foundUser.id });
```

Comment:
Ne devrait-on pas ajouter une nouvelle méthode pour récupérer un élément de type `authentication-method` grâce à son **userId** et **identityProvider** ?

Je partirai plutôt vers une méthode générique, si possible, comme `findBy` qui recevrait un objet en paramètre `{ userId, indentityProvider, ...rest }`.

```suggestion
  const authenticationMethod = await authenticationMethodRepository.findBy({ userId: foundUser.id, identityProvider });
```

Reply:
Il existe cette méthode `findOneByUserIdAndIdentityProvider` qui répondrait à ta suggestion :)

Reply:
done !